app/main.py

```diff
@@ -1 +1,41 @@
-# write your code here
+from __future__ import annotations
+
+
+class Animal:
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.hidden = hidden
+        self.health = health
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (
+            f"{{Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        if self.hidden:
+            setattr(self, "hidden", False)
+        elif not self.hidden:
+            setattr(self, "hidden", True)
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(herbivore: Herbivore) -> None:
+        if not herbivore.hidden and not isinstance(herbivore, Carnivore):
+            herbivore.health -= 50
+        for num in range(len(Animal.alive)):
```

Comment:
There no need to iterate over whole 'Animal.alive'
Just check if self is present in alive and remove if is